Zend/zend_inheritance.c

```diff
@@ -2004,6 +2007,12 @@ static void zend_add_trait_method(zend_class_entry *ce, zend_string *name, zend_
 	function_add_ref(new_fn);
 	fn = zend_hash_update_ptr(&ce->function_table, key, new_fn);
 	zend_add_magic_method(ce, fn, key);
+
+	if (inherited && new_fn->common.fn_flags & ZEND_ACC_OVERRIDE) {
+		do_inheritance_check_on_method(
+				new_fn, new_fn->common.scope, existing_fn, existing_fn->common.scope,
+				new_fn->common.scope, NULL, /* check_visibility */ 1);
```

Comment:
Rather than performing the whole check again, we should only remove the `ZEND_ACC_OVERRIDE` flag.

Reply:
I've re-performed the whole check, because the check aborts early with a success, but without unsetting the OVERRIDE flag in some cases. I've now duplicated the necessary logic, but I'm not quite sure if this is any better.

I've added test cases for this behavior, though. So your remark is a win in any case :smile:

Reply:
You could also try delaying the inheritance check until we've copied the function so that the flag can be removed as normal.

Reply:
I have absolutely no idea what I'm doing, but it appears to have worked. All green and the code does not look too shabby 😁